src/mc_tasks/ImpedanceTask.cpp

```diff
@@ -0,0 +1,238 @@
+                             unsigned int robotIndex,
+                             double stiffness,
+                             double weight,
+                             double impM,
+                             double impD,
+                             double impK)
+: SurfaceTransformTask(surfaceName, robots, robotIndex, stiffness, weight)
+{
+  const auto & robot = robots.robot(robotIndex);
+  if(!robot.surfaceHasIndirectForceSensor(surfaceName))
+  {
+    mc_rtc::log::error_and_throw<std::runtime_error>(
+        "[mc_tasks::ImpedanceTask] Surface {} does not have a force sensor attached", surfaceName);
+  }
+
+  type_ = "impedance";
+  name_ = "impedance_" + robots.robot(rIndex).name() + "_" + surfaceName;
+
+  // Set impedance parameters
+  impedance(sva::ForceVecd(Eigen::Vector3d::Constant(10.0 * impM), Eigen::Vector3d::Constant(impM)),
+            sva::ForceVecd(Eigen::Vector3d::Constant(10.0 * impD), Eigen::Vector3d::Constant(impD)),
+            sva::ForceVecd(Eigen::Vector3d::Constant(10.0 * impK), Eigen::Vector3d::Constant(impK)));
+}
+
+void ImpedanceTask::update(mc_solver::QPSolver & solver)
+{
+  // 1. Compute the compliance acceleration
+  sva::PTransformd T_0_s(surfacePose().rotation());
+  // compAccelW_ is represented in the world frame
+  //   \ddot{p}_c = \ddot{p}_d + \frac{D}{M} (\dot{p}_d - \dot{p}_c) + \frac{K}{M} (p_d - p_c) + \frac{K_f}{M} (f_m -
+  //   f_d) See the Constructor description for the definition of symbols
+  compAccelW_ =
+      desiredAccelW_
+      + T_0_s.invMul( // T_0_s.invMul transforms the MotionVecd value from the surface frame to the world frame
+            sva::MotionVecd(
+                // Compute in the surface frame because the impedance parameters and wrench gain are represented in the
+                // surface frame
+                impM_.vector().cwiseInverse().cwiseProduct(
+                    // T_0_s transforms the MotionVecd value from the world from to the surface frame
+                    impD_.vector().cwiseProduct((T_0_s * (desiredVelW_ - compVelW_)).vector())
+                    + impK_.vector().cwiseProduct((T_0_s * transformError(compPoseW_, desiredPoseW_)).vector())
+                    + wrenchGain_.vector().cwiseProduct((measuredWrench() - targetWrench_).vector()))));
+
+  // 2. Compute the compliance pose and velocity by time integral
+  double dt = solver.dt();
+  // 2.1 Integrate velocity to pose
+  sva::PTransformd T_0_c(compPoseW_.rotation());
+  // Represent the compliance velocity and acceleration in the compliance frame and scale by dt
+  sva::MotionVecd compDeltaC = T_0_c * (dt * compVelW_ + 0.5 * dt * dt * compAccelW_);
+  // Convert the angular velocity to the rotation matrix through AngleAxis representation
+  Eigen::AngleAxisd aaCompDeltaC(Eigen::Quaterniond::Identity());
+  if(compDeltaC.angular().norm() > 1e-6)
+  {
+    aaCompDeltaC = Eigen::AngleAxisd(compDeltaC.angular().norm(), compDeltaC.angular().normalized());
+  }
```

Comment:
I think it would be nice to have a more direct way to do this kind of integration in `SpaceVecAlg`. I'm not sure how to best implement it though (ping @mehdi-benallegue)